src/bpf/bpfctl.c

```diff
@@ -0,0 +1,238 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <getopt.h>
+
+#include "build.h"
+#include "format-table.h"
+#include "format-util.h"
+#include "main-func.h"
+#include "parse-argument.h"
+#include "pretty-print.h"
+#include "varlink.h"
+#include "verbs.h"
+
+static JsonFormatFlags arg_json_format_flags = JSON_FORMAT_OFF;
+
+typedef struct BpfProgramData BpfProgramData;
+typedef struct BpfMapData BpfMapData;
+
+struct BpfProgramData {
+        char *name;
+        uint64_t type;
+        uint64_t memlock;
+        char *tag;
+        char **map_names;
+        uint64_t run_time_ns;
+        uint64_t run_cnt;
+};
+
+struct BpfMapData {
+        char *name;
+        uint64_t type;
+        uint64_t memlock;
+};
+
+static void bpf_prog_data_done(BpfProgramData *p) {
```

Comment:
please name the functions just like the types they destroy, i.e. bpf_program_data_done() here. those three characters won't hurt